android/src/main/java/com/amplitude/android/plugins/AndroidNetworkConnectivityCheckerPlugin.kt

```diff
@@ -0,0 +1,50 @@
+package com.amplitude.android.plugins
+
+import AndroidNetworkListener
+import com.amplitude.android.Configuration
+import com.amplitude.android.utilities.AndroidNetworkConnectivityChecker
+import com.amplitude.core.Amplitude
+import com.amplitude.core.platform.Plugin
+import kotlinx.coroutines.launch
+
+class AndroidNetworkConnectivityCheckerPlugin : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Before
+    override lateinit var amplitude: Amplitude
+    internal var networkConnectivityChecker: AndroidNetworkConnectivityChecker? = null
+    internal var networkListener: AndroidNetworkListener? = null
+
+    override fun setup(amplitude: Amplitude) {
+        super.setup(amplitude)
+        if (amplitude.configuration.offline != null) {
+            amplitude.logger.debug("Installing AndroidNetworkConnectivityPlugin, offline feature should be supported.")
+            networkConnectivityChecker = AndroidNetworkConnectivityChecker((amplitude.configuration as Configuration).context, amplitude.logger)
+            amplitude.amplitudeScope.launch(amplitude.storageIODispatcher) {
+                amplitude.configuration.offline = !networkConnectivityChecker!!.isConnected()
+            }
+            val networkChangeHandler =
+                object : AndroidNetworkListener.NetworkChangeCallback {
+                    override fun onNetworkAvailable() {
+                        amplitude.logger.debug("AndroidNetworkListener, onNetworkAvailable.")
+                        amplitude.configuration.offline = false
+                        amplitude.flush()
+                    }
+
+                    override fun onNetworkUnavailable() {
+                        amplitude.logger.debug("AndroidNetworkListener, onNetworkUnavailable.")
+                        amplitude.configuration.offline = true
+                    }
+                }
+            networkListener = AndroidNetworkListener((amplitude.configuration as Configuration).context)
+            networkListener!!.setNetworkChangeCallback(networkChangeHandler)
+            networkListener!!.startListening()
+        }
+    }
+
+    override fun teardown() {
+        if (amplitude.configuration.offline != null) {
```

Comment:
seems a workaround to disable other than remove. Remove is a bit more preferred though the hackaround might be ok. Wonder if we still need this check here.

Reply:
I agree I think I prefer the `amplitude.remove("network-connectivity-detector")`.

I'm guessing most people will leave this as-is.

Also remove Plugin by name would be helpful for other use cases and to match the Browser SDK functionality.

Reply:
I'd push back on this.. Remove by name seems to be more like a feature request and is out of the scope of offline feature. Disabling offline by `config.offline = null` also aligns with how we disable default tracking by `config.defaultTracking = DefaultTrackingOptions.NONE`.